network/tor/tor/src/main/java/bisq/network/tor/process/control_port/ControlPortFilePoller.java

```diff
@@ -45,26 +48,94 @@ private void startPoller() {
         Thread thread = new Thread(() -> {
```

Comment:
Consider using `ExecutorFactory.newSingleThreadScheduledExecutor("ControlPortFilePollerThread")`, keeping a class reference and add a shutdown method to stop the executorService.